tests/FSharp.Compiler.ComponentTests/EmittedIL/Literals.fs

```diff
@@ -176,36 +176,95 @@ let [<Literal>] x = System.Int32.MaxValue + 1
         }
 
     [<Fact>]
-    let ``Compilation fails when using decimal arithmetic in literal``() =
+    let ``Arithmetic can be used for constructing decimal literals``() =
         FSharp """
 module LiteralArithmetic
 
-let [<Literal>] x = 1m + 1m
+[<Literal>]
+let x = 1m + 2m
         """
         |> withLangVersion80
         |> compile
-        |> shouldFail
-        |> withResults [
-            { Error = Error 267
-              Range = { StartLine = 4
-                        StartColumn = 21
-                        EndLine = 4
-                        EndColumn = 23 }
-              Message = "This is not a valid constant expression or custom attribute value" }
-            { Error = Error 267
-              Range = { StartLine = 4
-                        StartColumn = 26
-                        EndLine = 4
-                        EndColumn = 28 }
-              Message = "This is not a valid constant expression or custom attribute value" }
-            { Error = Error 267
-              Range = { StartLine = 4
-                        StartColumn = 21
-                        EndLine = 4
-                        EndColumn = 28 }
-              Message = "This is not a valid constant expression or custom attribute value" }
+        |> shouldSucceed
+        |> verifyIL [
+            """.field public static initonly valuetype [runtime]System.Decimal x"""
+            """.custom instance void [runtime]System.Runtime.CompilerServices.DecimalConstantAttribute::.ctor(uint8,
```

Comment:
Could you also please add an explicit baseline that a cctor (static one) is codegened (that the init code goes after .cctor is emitted) for the type we emit field for, and manually verify that C# is generating roughly the same IL (if you didn't do so already)? Thanks!

Reply:
I think this works correctly.
But indeed, would be more explicit if the test's assertion included the method definition for the cctor to make it visible when this gets init'd.

Here is the C# IL, looks the same:
https://sharplab.io/#v2:C4LglgNgPgAgzAAhgJgQYQQbwLACgEEIDGA9gHYDOwCAJgKZFgC2AhhAgLICeAcgK5MARnQBOCALwIAjADpkAVgBsAdiYBuPAF8gA===

Reply:
I think it's slightly different than the IL C# is generating because the init of the decimal happens inside a `staticInitialization` method which in turn is called from the static `.cctor`.
So for
```fsharp
module DecimalInit
[<Literal>]
let x = 5.5m
```
it generates
```
.class public abstract auto ansi sealed DecimalInit
       extends [runtime]System.Object
{
  .field public static initonly valuetype [runtime]System.Decimal x
  .custom instance void [runtime]System.Runtime.CompilerServices.DecimalConstantAttribute::.ctor(uint8,
                                                                                                        uint8,
                                                                                                        int32,
                                                                                                        int32,
                                                                                                        int32) = ( 01 00 01 00 00 00 00 00 00 00 00 00 37 00 00 00   
                                                                                                                   00 00 ) 
  .method private specialname rtspecialname static void  .cctor() cil managed
  {
    
    .maxstack  8
    IL_0000:  ldc.i4.0
    IL_0001:  stsfld     int32 '<StartupCode$assembly>'.$DecimalInit::init@
    IL_0006:  ldsfld     int32 '<StartupCode$assembly>'.$DecimalInit::init@
    IL_000b:  pop
    IL_000c:  ret
  } 

  .method assembly specialname static void staticInitialization@() cil managed
  {
    
    .maxstack  8
    IL_0000:  ldc.i4.s   55
    IL_0002:  ldc.i4.0
    IL_0003:  ldc.i4.0
    IL_0004:  ldc.i4.0
    IL_0005:  ldc.i4.1
    IL_0006:  newobj     instance void [runtime]System.Decimal::.ctor(int32,
                                                                             int32,
                                                                             int32,
                                                                             bool,
                                                                             uint8)
    IL_000b:  stsfld     valuetype [runtime]System.Decimal DecimalInit::x
    IL_0010:  ret
  } 

} 

.class private abstract auto ansi sealed '<StartupCode$assembly>'.$DecimalInit
       extends [runtime]System.Object
{
  .field static assembly int32 init@
  .method private specialname rtspecialname static void  .cctor() cil managed
  {
    
    .maxstack  8
    IL_0000:  call       void DecimalInit::staticInitialization@()
    IL_0005:  ret
  } 

}
```
I do agree it could be more clear from the tests and I'll update them accordingly if the IL is acceptable.

Reply:
That is fine, both end up calling the same sequence of instructions eventually.

Reply:
Yeah, I checked the emitted IL, it looks correct, just wanted it to be reflected in the baseline, so if we ever make any other changes to static initialization, there's higher chance of catching regression.

Thanks for adding it!

Reply:
I guess we didn't update the test :(

Reply:
No, sorry I did not have time yet to update the tests. Would it be OK to add those in a separate PR (within a couple days)? Otherwise feel free to revert.

Reply:
Not a problem. @esbenbjerre , please update the assertion in way which makes the callchain from cctor explicit in a separate PR. No need to revert.